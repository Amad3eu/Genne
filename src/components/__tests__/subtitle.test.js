import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Subtitle from '../subtitle';

describe('Subtitle', () => {
  const mockEvent = { target: { value: 'This is a mock event' } };
  const props = {
    data: {
      subtitle: 'A frontend developer',
    },
    handleDataChange: jest.fn().mockReturnValue({}),
  };

  const component = shallow(<Subtitle {...props} />);

  it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('calls onChange', () => {
    component.find('input').at(0).simulate('change', mockEvent);
    expect(props.handleDataChange).toBeCalledWith('subtitle', mockEvent);
  });
});
