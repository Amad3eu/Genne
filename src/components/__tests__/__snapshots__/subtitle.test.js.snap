// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle renders correctly 1`] = `
<div
  className="flex justify-center items-start flex-col w-full px-2 sm:px-6 mb-10"
>
  <div
    className="text-xl sm:text-2xl font-bold font-title mt-2 mb-2"
  >
    Subtitle
  </div>
  <input
    className="outline-none w-full text-xs sm:text-lg sm:w-1/2 border-t-0 border-l-0 border-r-0 border solid border-gray-900 py-1 px-2 focus:border-blue-700"
    id="subtitle"
    onChange={[Function]}
    value="A frontend developer"
  />
</div>
`;
